extensions/api/data-management/contractagreement/src/main/java/org/eclipse/dataspaceconnector/api/datamanagement/contractagreement/ContractAgreementApiExtension.java

```diff
@@ -9,26 +9,56 @@
  *
  *  Contributors:
  *       Microsoft Corporation - initial API and implementation
+ *       Bayerische Motoren Werke Aktiengesellschaft (BMW AG) - add functionalities
  *
  */
 
 package org.eclipse.dataspaceconnector.api.datamanagement.contractagreement;
 
 import org.eclipse.dataspaceconnector.api.datamanagement.configuration.DataManagementApiConfiguration;
+import org.eclipse.dataspaceconnector.api.datamanagement.contractagreement.service.ContractAgreementServiceImpl;
+import org.eclipse.dataspaceconnector.api.datamanagement.contractagreement.transform.ContractAgreementToContractAgreementDtoTransformer;
+import org.eclipse.dataspaceconnector.api.transformer.DtoTransformerRegistry;
 import org.eclipse.dataspaceconnector.spi.WebService;
+import org.eclipse.dataspaceconnector.spi.contract.negotiation.store.ContractNegotiationStore;
 import org.eclipse.dataspaceconnector.spi.system.Inject;
 import org.eclipse.dataspaceconnector.spi.system.ServiceExtension;
 import org.eclipse.dataspaceconnector.spi.system.ServiceExtensionContext;
+import org.eclipse.dataspaceconnector.spi.transaction.NoopTransactionContext;
+import org.eclipse.dataspaceconnector.spi.transaction.TransactionContext;
+
+import static java.util.Optional.ofNullable;
 
 public class ContractAgreementApiExtension implements ServiceExtension {
+
     @Inject
-    private WebService webService;
+    WebService webService;
 
     @Inject
-    private DataManagementApiConfiguration config;
+    DataManagementApiConfiguration config;
+
+    @Inject
+    DtoTransformerRegistry transformerRegistry;
+
+    @Inject(required = false)
+    TransactionContext transactionContext;
+
+    @Inject
+    ContractNegotiationStore store;
 
     @Override
     public void initialize(ServiceExtensionContext context) {
-        webService.registerResource(config.getContextAlias(), new ContractAgreementApiController(context.getMonitor()));
+        transformerRegistry.register(new ContractAgreementToContractAgreementDtoTransformer());
+        var monitor = context.getMonitor();
+
+        var transactionContextImpl = ofNullable(transactionContext)
```

Comment:
Another FYI: once we have the `@Provides(isDefault=true)` in place, we should change this to a required `@Inject`